### PR TITLE
feat: add command to run opened chk file

### LINKED ISSUE #29: Add command and shortcut to run currently opened `.chk` file in a new terminal instance
We need to know following things:

- [x] #30
- [x] #31

### LINKED ISSUE #30: Add a external task, with a shortcut of our choice

### LINKED ISSUE #31: Creates and usage separate terminal window when we run the task

### PR DESCRIPTION
**Pull request by arifszn**
Close #30.
Close #31.

The command can be run from the command palette or by keyboard shortcut: `alt+ctrl+r`. Should the keybinding be changed?

Also, not tested on Mac.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -39,6 +40,18 @@
...
+      {
+        "command": "chkware.runFile",
+        "title": "Run file",
+        "category": "chkware"
+      }
+    ],
+    "keybindings": [
+      {
+        "command": "chkware.runFile",
+        "key": "ctrl+i ctrl+i",
```
Comment (0hsn) [inline package.json]:
```suggestion
        "key": "alt+ctrl+r",
```

File: package.json [hunk 2]
```diff
@@ -39,6 +40,18 @@
...
+        "mac": "cmd+i cmd+i"
```
Comment (0hsn) [inline package.json]:
```suggestion
        "mac": "opt+ctrl+r"
```

File: src/commands/run-file.ts [hunk 3]
```diff
@@ -0,0 +1,65 @@
+import * as vscode from "vscode";
+import * as path from "path";
+
+export async function runFile(): Promise<void> {
+  const activeTextEditor: vscode.TextEditor | undefined =
+    vscode.window.activeTextEditor;
+
+  if (!activeTextEditor) {
+    vscode.window.showWarningMessage("Please open a chk file first.");
```
Comment (0hsn) [inline src/commands/run-file.ts]:
```suggestion
    vscode.window.showWarningMessage("Please open a .chk file first.");
```

Does it support markdown? then:

```suggestion
    vscode.window.showWarningMessage("Please open a `.chk` file first.");
```

Comment (0hsn) [reply src/commands/run-file.ts]: would it be possible to enable this key or dropdown item from command-pallet only when a .chk file is open in editor section? .. just asking

File: src/commands/run-file.ts [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+  const { openedFileName, openedFolder } = getFileInfo(
+    activeTextEditor.document
+  );
+
+  const commandType = getCommandType(activeTextEditor.document.getText());
+
+  // check file extension and command type
+  if (!openedFileName.endsWith(".chk") || !commandType) {
+    vscode.window.showWarningMessage("Not a valid chk file to run.");
```
Comment (0hsn) [inline src/commands/run-file.ts]:
```suggestion
    vscode.window.showWarningMessage("Not a valid .chk file to run.");
```

**Review by 0hsn — CHANGES_REQUESTED** [review]

File: src/commands/run-file.ts (hunk 3 above)
Comment (arifszn) [reply src/commands/run-file.ts]: Yes possible. Added in https://github.com/chkware/vscode-ext/pull/42/commits/39b3156ee2e1699e22f28e3852596ae5653f8fa1.

**Comment by arifszn** [comment]
Changed have been made according to the reviews.

**Review by 0hsn — COMMENTED** [review]
I couldn't run it. Therefore need your help for local test on MacOS. 🙏

**Comment by arifszn** [comment]
> I couldn't run it. Therefore need your help for local test on MacOS. 🙏

What problem are you facing? Are you following [this guide](https://chkware.github.io/setup/setup-ext-dev) to run it?

**Comment by 0hsn** [comment]
> > I couldn't run it. Therefore need your help for local test on MacOS. 🙏
> 
> What problem are you facing? Are you following [this guide](https://chkware.github.io/setup/setup-ext-dev) to run it?


> What problem are you facing?

- When if press `f5` the extension gets loaded in debug mode.
- I then press `cmd + shift + p` to open command pallet - which works okay, then
    1) I open a new .chk snippet, DO NOT save it, and hit `ctrl + i, ctrl + i` but it do not run
    2) I open a new .chk snippet, THEN save it, and hit `ctrl + i, ctrl + i` but it do not run

> Are you following [this guide](https://chkware.github.io/setup/setup-ext-dev) to run it?

Yes did so!

Thanks

**Comment by arifszn** [comment]
> > > I couldn't run it. Therefore need your help for local test on MacOS. 🙏
> > 
> > 
> > What problem are you facing? Are you following [this guide](https://chkware.github.io/setup/setup-ext-dev) to run it?
> 
> > What problem are you facing?
> 
> * When if press `f5` the extension gets loaded in debug mode.
> * I then press `cmd + shift + p` to open command pallet - which works okay, then
>   
>   1. I open a new .chk snippet, DO NOT save it, and hit `ctrl + i, ctrl + i` but it do not run
>   2. I open a new .chk snippet, THEN save it, and hit `ctrl + i, ctrl + i` but it do not run
> 
> > Are you following [this guide](https://chkware.github.io/setup/setup-ext-dev) to run it?
> 
> Yes did so!
> 
> Thanks

The shortcut key has changed in favor of https://github.com/chkware/vscode-ext/pull/42#discussion_r1070513157. PR description has been updated.

**Comment by 0hsn** [comment]
> > > > I couldn't run it. Therefore need your help for local test on MacOS. 🙏
> > > 
> > > 
> > > What problem are you facing? Are you following [this guide](https://chkware.github.io/setup/setup-ext-dev) to run it?
> > 
> > 
> > > What problem are you facing?
> > 
> > 
> > 
> > * When if press `f5` the extension gets loaded in debug mode.
> > * I then press `cmd + shift + p` to open command pallet - which works okay, then
> >   
> >   1. I open a new .chk snippet, DO NOT save it, and hit `ctrl + i, ctrl + i` but it do not run
> >   2. I open a new .chk snippet, THEN save it, and hit `ctrl + i, ctrl + i` but it do not run
> > 
> > > Are you following [this guide](https://chkware.github.io/setup/setup-ext-dev) to run it?
> > 
> > 
> > Yes did so!
> > Thanks
> 
> The shortcut key has changed in favor of [#42 (comment)](https://github.com/chkware/vscode-ext/pull/42#discussion_r1070513157). PR description has been updated.

Unfortunatly shortcut do not works for me. Then I further investigated and found, no shortcut was registered at all for chkware. So, how do the extension register this automatically?

FWIW: I tried `npm install`, then `npm run watch`, then `f5` then `cmd+shift+p`, and search for run. This time the command runs. but, not with shortcut.

image

Also, in case of when I tried with other file extension, the run command goes away!

File: package.json (hunk 2 above)
Comment (arifszn) [reply package.json]: Should it be `opt+cmd+r`?

**Comment by arifszn** [comment]
- > Unfortunatly shortcut do not works for me.

Can you please try now?

- > Also, in case of when I tried with other file extension, the run command goes away!

Shouldn't it be like this as discussed on https://github.com/chkware/vscode-ext/pull/42#discussion_r1070513589?

**Comment by 0hsn** [comment]
Awesome. This works for me now! Good work. 💟 

> Shouldn't it be like this as discussed on https://github.com/chkware/vscode-ext/pull/42#discussion_r1070513589?

I thought the option is going to stay, in disabled view, rather vanished at all. However, it's okay as well for now. We'll check this capability later. 🙏

**Review by 0hsn — APPROVED** [review]

**Pull request merged**